### ISSUE TITLE
Example of why we need to pin?

### ISSUE THREAD

**Issue by rsignell-usgs** [pangeo-data/pangeo-cloud-federation#5]
I tried my new non-pinned environment on dev.pangeo.io, and when I try to start a cluster, I get:
```python-traceback
distributed.core - ERROR - add_worker() got an unexpected keyword argument 'cpu'
Traceback (most recent call last):
  File "/srv/conda/lib/python3.6/site-packages/distributed/core.py", line 340, in handle_comm
    result = yield result
  File "/srv/conda/lib/python3.6/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/srv/conda/lib/python3.6/site-packages/tornado/gen.py", line 307, in wrapper
    result = func(*args, **kwargs)
```
So I guess the latest conda-forge packages `distributed=1.23.3` and  `tornado=5.1.1` don't play nice together, right?

So I should retreat to `distributed=1.22.1` and `tornado=5.0.2` and try again, right?

Just checking this is the right approach.  I won't ask this every time, I promise.

**Comment by mrocklin** [comment]
Distributed and tornado are fast friends and don't have any conflicts that
I'm aware of (I actually test against torando 6 (in development) from time
to time).  My first guess would be a version mismatch between different
distributed versions, but that shouldn't be happening with the current
setup unless you've pip installed something locally.

**Comment by mrocklin** [comment]
Everything worked for me:
https://github.com/pangeo-data/example.pangeo.io-deploy

**Comment by mrocklin** [comment]
Err, whoops, meant this link:
https://gist.github.com/798bedf21e5b729561d77cec2d19144d

**Comment by rsignell-usgs** [comment]
@mrocklin , I'm still getting:
```
 distributed.core - ERROR - add_worker() got an unexpected keyword argument 'cpu'
```
when I try to start a kube cluster.  

But after going back and forth with pinning, restarting the server, and getting super confused, I walked down the hall and had a colleague try dev.pangeo.io and it worked fine.   

So I checked my `~/.dask/config.yaml` and mine is pulling
```
image: us.gcr.io/pangeo-181919/dev-pangeo-io-notebook:d03a8f7
```
while his is pulling
```
image: us.gcr.io/pangeo-181919/dev-pangeo-io-notebook:a75c176
```

**Comment by jhamman** [comment]
IIRC, this error is fixed in dask-kubernetes master. I think you should be able to use the latest version of distributed with master.

**Comment by mrocklin** [comment]
OK, so maybe you had a config file from before and as we updated the image
your config file wasn't replaced.

@dsludwig what happens when there is a conflict between a file in the
user's environment and the image?  Do we overwrite?

**Comment by dsludwig** [comment]
Ahh... it uses the one in the image, because I didn't want to clobber work that users had done on any examples. Perhaps it needs to take the newer one instead (so if the config gets updated or an example gets updated, it will clobber the existing one).

An alternative is to change the startup script that rewrites the `image` in `.dask/config.yaml` to still work when there's a name in there.

**Comment by jhamman** [comment]
closed via #15
